src/Filters/FiltersOperator.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Spatie\QueryBuilder\Filters;
+
+use Illuminate\Database\Eloquent\Builder;
+use Spatie\QueryBuilder\Enums\FilterOperator;
+
+/**
+ * @template TModelClass of \Illuminate\Database\Eloquent\Model
+ * @template-implements \Spatie\QueryBuilder\Filters\Filter<TModelClass>
+ */
+class FiltersOperator extends FiltersExact implements Filter
+{
+    public function __construct(protected bool $addRelationConstraint, protected FilterOperator $filterOperator, protected string $boolean)
+    {
+    }
+
+    /** {@inheritdoc} */
+    public function __invoke(Builder $query, $value, string $property)
+    {
+        $filterOperator = $this->filterOperator;
+
+        if ($this->addRelationConstraint) {
+            if ($this->isRelationProperty($query, $property)) {
+                $this->withRelationConstraint($query, $value, $property);
+
+                return;
+            }
+        }
+
+        if (is_array($value)) {
+            $query->where(function ($query) use ($value, $property) {
+                foreach($value as $item) {
+                    $this->__invoke($query, $item, $property);
+                }
+            });
+
+            return;
+        }
+        else if ($this->filterOperator->isDynamic()) {
+            $filterOperator = $this->getDynamicFilterOperator($value, $this);
+            $this->removeDynamicFilterOperatorFromValue($value, $filterOperator);
+        }
+
+        $query->where($query->qualifyColumn($property), $filterOperator->value, $value, $this->boolean);
+    }
+
+    protected function getDynamicFilterOperator(string $value): FilterOperator
+    {
+        $filterOperator = FilterOperator::EQUAL;
+
+        // match filter operators and assign the filter operator.
+        foreach(FilterOperator::cases() as $filterOperatorCase) {
+            if (str_starts_with($value, $filterOperatorCase->value) && ! $filterOperatorCase->isDynamic()) {
```

Comment:
I've had a play around with this PR and this line means that `>` will always be returned instead of `>=` and the same for `<` instead of `<=`.

I think there's maybe an opportunity to include functionality for things like null/not null in this PR too perhaps?

Reply:
Depending on the sort that i made for the FilterOperator's cases what you say will not occur because >= will override >